logstash-core/spec/logstash/agent/converge_spec.rb

```diff
@@ -73,19 +73,21 @@
 
         describe "#running_user_defined_pipelines" do
           it "returns the user defined pipelines" do
-            start_agent(subject)
+            agent_task = start_agent(subject)
             wait_for do
               subject.with_running_user_defined_pipelines {|pipelines| pipelines.keys }
             end.to eq([:main])
+            agent_task.stop!
           end
         end
 
         describe "#running_user_defined_pipelines?" do
           it "returns true" do
-            start_agent(subject)
+            agent_task = start_agent(subject)
             wait_for do
               subject.running_user_defined_pipelines?
             end.to be_truthy
+            agent_task.stop!
```

Comment:
We should move the `start_agent` and `agent_task.stop!` lines to rspec `before/after(:each)` blocks. This will ensure that even if an exception is thrown `agent_task.stop!` will be called. This is equivalent to an `ensure`.

Reply:
Done.